### ISSUE TITLE
Change default value for dist= in EndMissionTake and AudioChange

### ISSUE THREAD

**Issue by krzys-h** [colobot/colobot#759]
***WARNING! This is a backwards incompatible change that WILL break user levels that don't specify the value explicitly***

The default value for dist= has always confused me. By default it's set to value appropariate to cover the whole SpaceShip. If I omit pos= and dist=, I would expect the command to detect anywhere on the level. This is not how it works.

Consider this:
```
EndMissionTake type=WheeledGrabber min=1
```
This looks like "the mission ends when one WheeledGrabber is produced anywhere on the map", while actually this is "the mission ends when one WheeledGrabber is placed at max 8 meters from the center of the map"
To get the desired behaviour, you have to type:
```
EndMissionTake dist=25000.00 type=WheeledGrabber min=1
```
Which is longer and confusing.

---

Another common misconception might be assuming that pos= defaults to SpaceShip position. This is not how it works either. pos= defaults to the center of the map. To check for objects on SpaceShip, you still have to do this:
```
EndMissionTake pos=x;y type=TitaniumOre min=4 // dist= omitted to keep the default value
```
I believe it would be better to make this more explicit:
```
EndMissionType pos=x;y dist=8 type=TitaniumOre min=4 // dist= is required to detect around the position
```

---

Additionaly, using only pos or only dist should not be allowed to avoid mistakes and the level parser should throw an error.

I would like to know what other people think about changing this, especially (user)level designers.

**Comment by tomaszkax86** [comment]
This is definitely a better version, less confusing and it makes more sense than current defaults. We might want to implement this, but we'll have to review existing missions.

**Comment by RaptorParkowsky** [comment]
BTW why we still care about backward compatibility?

**Comment by MrSimbax** [comment]
The same reason we're not changing the CBOT design, it's a remake, not Colobot 2.

**Comment by RaptorParkowsky** [comment]
Of course, it's a remake but in the plot/design/gameplay aspects, not engine aspects.

**Comment by tomangelo2** [comment]
We did already things breaking backward compatibility with original game, so why don't break compatibility with early alpha versions?

**Comment by MrSimbax** [comment]
You didn't understand, there are already user levels made specifically for GOLD, and this change will break backwards compatibility with them. That's why it's important to discuss it especially that it's a change that is easy to miss.

**Comment by RaptorParkowsky** [comment]
That's why I always wanted inform level creators that they should do readme's for their work with info about what version of the GOLD is required. Until the game isn't finished, especially Scene refactoring, they should unfortunately live with that.

**Comment by krzys-h** [comment]
Planned roadmap for this issue:
* [x] 0.1.8: Trying to load a level with pos= or dist= will generate a user-visible warning to force userlevel authors to update the levels. Old levels will still load correctly.
* [ ] 0.1.9: The defaults will change. Old levels will either throw parser errors or cause strange behaviour in ending conditions.

New (planned) defaults are:
* dist= - infinity (whole map)
* pos= - no default value (if dist= is specified then pos= is required, otherwise it is ignored and trying to specify it will cause an error)

If you want to keep your levels compatible with both old and new versions, always specify both pos= and dist=. If you only care about 0.1.9+, you can skip both of them to detect objects on the whole map.
